### PR TITLE
Font: document new font features

### PR DESCRIPTION
**Pull request by clydebarrow**
## Description:

**Related issue (if applicable):** fixes <link to issue>

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#6198

## Checklist:

  - [x] I am merging into `next` because this is new documentation that has a matching pull-request in [esphome](https://github.com/esphome/esphome) as linked above.  
    or
  - [ ] I am merging into `current` because this is a fix, change and/or adjustment in the current documentation and is not for a new component or feature.

  - [ ] Link added in `/index.rst` when creating new documents for new components or cookbook.

### CODE REVIEW

File: components/display/index.rst [hunk 1]
```diff
@@ -9,10 +9,10 @@ The ``display`` component houses ESPHome's powerful rendering and display
 engine. Fundamentally, there are these types of displays:
 
 - Text based displays like :doc:`7-Segment displays <max7219>` or
-  :doc:`some LCD displays <lcd_display>`.
-- Displays like the :doc:`nextion` that have their own processors for rendering.
-- Binary displays which can toggle ON/OFF any pixel, like :doc:`E-Paper displays <waveshare_epaper>` or
-  :doc:`OLED displays <ssd1306>`.
+  :doc:`LCD displays <lcd_display>`.
```
Comment (kbx81) [inline components/display/index.rst:12]: "Liquid Crystal Display displays"...stuff like this always triggers my OCD. 😅

File: components/display/fonts.rst [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+          - file: "fonts/materialdesignicons-webfont.ttf"
+            glyphs: [
+              "\U000F02D1", # mdi-heart
+              "\U000F05D4", # mdi-airplane-landing
+              ]
+
+    display:
+      # ...
+
+Configuration options:
```
Comment (kbx81) [inline components/display/fonts.rst]: I think normally we use:
```suggestion
Configuration variables:
```

File: components/display/index.rst (hunk 1 above)
Comment (clydebarrow) [reply components/display/index.rst:12]: In my defence, that was already there, just got moved a bit :-)

Comment (kbx81) [reply components/display/index.rst:12]: Haha yes indeed, I noticed 😅

**Review by kbx81 — APPROVED** [review]

**Pull request merged**